NuRadioReco/modules/io/coreas/readCoREASDetector.py

```diff
@@ -0,0 +1,330 @@
+
+    def begin(self, input_files, xmin=0, xmax=0, ymin=0, ymax=0, n_cores=10, seed=None, core_position_list=[], selected_station_ids=[], selected_channel_ids=[], interp_lowfreq=30*units.MHz, interp_highfreq=1000*units.MHz, log_level=logging.INFO, debug=False):
+            
+        """
+        begin method
+        initialize readCoREAS module
+        Parameters
+        ----------
+        input_files: input files
+            list of coreas hdf5 files
+        xmin: float
+            minimum x coordinate of the area in which core positions are distributed
+        xmax: float
+            maximum x coordinate of the area in which core positions are distributed
+        ymin: float
+            minimum y coordinate of the area in which core positions are distributed
+        ynax: float
+            maximum y coordinate of the area in which core positions are distributed
+        n_cores: number of cores (integer)
+            the number of random core positions to generate for each input file
+        seed: int (default: None)
+            Seed for the random number generation. If None is passed, no seed is set
+        interp_lowfreq: float (default = 30)
+            lower frequency for the bandpass filter in interpolation
+        interp_highfreq: float (default = 1000)
+            higher frequency for the bandpass filter in interpolation
+        """
+        self.__input_files = input_files
+        self.__n_cores = n_cores
+        self.__current_input_file = 0
+        self.__area = [xmin, xmax, ymin, ymax]
+        self.__core_position_list = core_position_list
+        self.__interp_lowfreq = interp_lowfreq
+        self.__interp_highfreq = interp_highfreq
+        self.__selected_station_ids = selected_station_ids
+        self.__selected_channel_ids = selected_channel_ids
+        self.__random_generator = np.random.RandomState(seed)
+        self.logger.setLevel(log_level)
+        self.debug = debug
+
+    def get_random_core_positions(self):
+        # generate core positions randomly within a rectangle
+        cores = np.array([self.__random_generator.uniform(self.__area[0], self.__area[1], self.__n_cores),
+                        self.__random_generator.uniform(self.__area[2], self.__area[3], self.__n_cores),
+                        np.zeros(self.__n_cores)]).T
+        return cores
+
+    def get_interpolated_efield(self, position, core):
+        """
+        Accesses the interpolated electric field at the position of the detector on ground. Set pulse_centered to True to
+        shift all pulses to the center of the trace and account for the physical time delay of the signal.
+        """
+        antenna_position = position
+        antenna_position[2] = 0
+        # transform antenna position into shower plane with respect to core position, core position is set to 0,0 in shower plane
+        antenna_pos_vBvvB = self.cs.transform_to_vxB_vxvxB(antenna_position, core=core)
```

Comment:
I (possibly mis-)understood the antenna z position should be set to zero in shower plane coordinates, and not ground coordinates. Here it seems to be done in ground coordinates. 
```suggestion
        # transform antenna position into shower plane with respect to core position, core position is set to 0,0 in shower plane
        antenna_pos_vBvvB = self.cs.transform_to_vxB_vxvxB(antenna_position, core=core)
        antenna_pos_vBvvB[2] = 0
```

Reply:
It was indeed intended to be on the ground coordinates, such that the core positions which were only given in x and y are in the same z-plane as the antenna positions for the transformation. I changed it now to have the same value as the core z positions. For the distance to shower core in the shower plane only x and y are used, same for the interpolation.